OTAnalytics/plugin_ui/customtkinter_gui/frame_filter.py

```diff
@@ -0,0 +1,503 @@
+            validatecommand=(self.register(self._validate_date_format), "%P", "%W"),
+        )
+        self.hour_entry = CTkEntry(
+            master=self,
+            placeholder_text="HH",
+            width=35,
+            validate="key",
+            validatecommand=(
+                self.register(self._validate_time_format),
+                HOUR,
+                "%P",
+                "%W",
+            ),
+        )
+        self.minute_entry = CTkEntry(
+            master=self,
+            placeholder_text="MM",
+            width=35,
+            validate="key",
+            validatecommand=(
+                self.register(self._validate_time_format),
+                MINUTE,
+                "%P",
+                "%W",
+            ),
+        )
+        self.second_entry = CTkEntry(
+            master=self,
+            placeholder_text="SS",
+            width=35,
+            validate="key",
+            validatecommand=(
+                self.register(self._validate_time_format),
+                SECOND,
+                "%P",
+                "%W",
+            ),
+        )
+
+    def _place_widgets(self) -> None:
+        self.name_label.grid(row=0, column=0, padx=PADX, pady=PADY, sticky=STICKY_WEST)
+        self.date_entry.grid(row=0, column=1, padx=PADX, pady=PADY, stick=STICKY)
+        self.hour_entry.grid(row=0, column=2, padx=PADX, pady=PADY, stick=STICKY)
+        ColonLabel(master=self).grid(row=0, column=3)
+        self.minute_entry.grid(row=0, column=4, padx=PADX, pady=PADY, stick=STICKY)
+        ColonLabel(master=self).grid(row=0, column=5)
+        self.second_entry.grid(row=0, column=6, padx=PADX, pady=PADY, stick=STICKY)
+        self.validation_info_label.grid(
+            row=0, column=7, padx=PADX, pady=PADY, sticky=STICKY
+        )
+
+    def _display_text_on_entry_widget(self, widget: CTkEntry, text: str) -> None:
+        widget.delete(0, END)
+        widget.insert(0, text)
+
+    def _validate_date_format(self, value: str, widget_name: str) -> bool:
+        widget: CTkEntry = self.nametowidget(widget_name).master
+        if self._viewmodel.validate_date(value):
+            widget.configure(border_color=COLOR_GREEN)
+            self._clear_validation_info()
+        else:
+            widget.configure(border_color=COLOR_RED)
+            self._display_invalid_validation_info(
+                f"Date must be of format: '{DATE_FORMAT_PLACEHOLDER}'"
+            )
+        return True
+
+    def _validate_time_format(self, name: str, value: str, widget_name: str) -> bool:
+        widget: CTkEntry = self.nametowidget(widget_name).master
+
+        if not self._has_valid_time_format(name, value):
+            self._display_invalid_validation_info(self._build_error_message_for(name))
+            widget.configure(border_color=COLOR_RED)
+            return True
+
+        widget.configure(border_color=COLOR_GREEN)
+        self._clear_validation_info()
+        return True
+
```

Comment:
The if structure in this method is semantically the same as in the `_valida_date_format` method: Validate a date or time object and show the result as color.

However, the logic in `_validate_date_format` is positive: `if self._viewmodel.validate_date(value):` which is easy to read. The logic in `_validate_time_format` is negative: `if not ...` which is harder to read.

Can you change the structure of the methods to be more or less the same?

Reply:
Will do!